client/client_grpc_options_test.go

```diff
@@ -13,59 +13,57 @@ package client
 
 import (
 	"context"
-	"testing"
-
 	"github.com/golang/protobuf/proto"
 	"github.com/milvus-io/milvus-sdk-go/v2/entity"
 	"github.com/milvus-io/milvus-sdk-go/v2/internal/proto/common"
 	"github.com/milvus-io/milvus-sdk-go/v2/internal/proto/server"
 	"github.com/stretchr/testify/assert"
-	tmock "github.com/stretchr/testify/mock"
-	"github.com/stretchr/testify/require"
+	"testing"
 )
 
-func TestCollectionWithConsistencyLevel(t *testing.T) {
-	opt := CollectionWithConsistencyLevel(entity.ConsistencyLevel(entity.CL_BOUNDED))
+func TestCreateCollectionWithConsistencyLevel(t *testing.T) {
+	opt := WithConsistencyLevel(entity.ConsistencyLevel(entity.CL_BOUNDED))
 	assert.NotNil(t, opt)
 	req := &server.CreateCollectionRequest{}
 
 	assert.NotPanics(t, func() {
-		opt.OptCreateCollection(req)
+		opt(req)
 	})
 
 	assert.Equal(t, common.ConsistencyLevel_Bounded, req.GetConsistencyLevel())
 }
 
-type mockCreateCollectionOpt struct {
-	tmock.Mock
-}
-
-func (m *mockCreateCollectionOpt) OptCreateCollection(req *server.CreateCollectionRequest) {
-	m.Called(req)
-}
-
-func TestCreateCollection_WithConsistencyLevel(t *testing.T) {
+func TestLoadCollectionWithReplicaNumber(t *testing.T) {
 	ctx := context.Background()
 	c := testClient(ctx, t)
-	// default, all collection name returns false
-	mock.delInjection(mHasCollection)
-
-	ds := defaultSchema()
-	shardsNum := int32(1)
-	opt := &mockCreateCollectionOpt{}
-	sch := ds.ProtoMessage()
-	bs, err := proto.Marshal(sch)
-	require.NoError(t, err)
-	req := &server.CreateCollectionRequest{
-		DbName:           "", // reserved fields, not used for now
-		CollectionName:   ds.CollectionName,
-		Schema:           bs,
-		ShardsNum:        shardsNum,
-		ConsistencyLevel: common.ConsistencyLevel_Strong,
-	}
-	opt.On("OptCreateCollection", req).Return()
+	mock.setInjection(mHasCollection, hasCollectionDefault)
+	defer mock.delInjection(mHasCollection)
+	t.Run("Load default replica", func(t *testing.T) {
+		mock.setInjection(mLoadCollection, func(ctx context.Context, raw proto.Message) (proto.Message, error) {
+			req, ok := raw.(*server.LoadCollectionRequest)
+			if !ok {
+				return badRequestStatus()
+			}
+			assert.Equal(t, testDefaultReplicaNumber, req.GetReplicaNumber())
+			assert.Equal(t, testCollectionName, req.GetCollectionName())
```

Comment:
This test case shall be in client_grpc_collection_test.go

Reply:
My bad to leave it here. Could you please move it to the correct file?

Reply:
Done. Please check again.